### PR TITLE
Overview3

### PR DESCRIPTION
**Pull request by EJLEE808**
Jaimie!

1. Merged changes from Github to my local
2. Clicked all merge incoming changes.

Added...
1. Overview Widget's Skeleton
2. Routes on Server w/ Helper Files
3. Spec File for Overview

Note: Did a "--no-verify for ESLint".

### CODE REVIEW

File: client/src/app.jsx [hunk 1]
```diff
@@ -2,8 +2,13 @@ import React from 'react';
 // eslint-disable-next-line no-unused-vars
 import ReactDOM from 'react-dom';
 import Main from './components/relatedProducts/Main.jsx';
+<<<<<<< HEAD
 import Overview from './components/overview/overview.jsx';
 import $ from 'jquery';
+=======
+// import $ from 'jquery';
+import RatingsAndReviews from './components/Reviews/Reviews.jsx';
+>>>>>>> 6115ae46967e5ad1adf303c873410a7b63c6625f
```
Comment (PBandJaimie) [inline client/src/app.jsx]: For some reason, the <<<<<Head thingy got left in here. It happened in my file too.

File: client/src/components/overview/overview.jsx [hunk 2]
```diff
@@ -23,11 +34,32 @@ class Overview extends React.Component {
...
   render() {
+    let renderFeatures;
+    let renderStyles;
+    if (this.state.info !== '') {
+      renderFeatures = <Features features={this.state.info.features} />
+      renderStyles = <Styles styles={this.state.styles.results} />
+    } else {
+      renderFeatures = <div></div>
+      renderStyles = <div></div>
+    }
```
Comment (PBandJaimie) [inline client/src/components/overview/overview.jsx:45]: Great use of conditional rendering here! I think maybe I was trying to use something like this last night and came across componentDidUpdate(). Now I'm wondering if conditional rendering is the better options. What do you think? https://gist.github.com/PBandJaimie/701bb5633e06917f5ba5d4aa56c98468#file-2021-07-28-md

**Review by PBandJaimie — CHANGES_REQUESTED** [review]
Okay! I think I got everything. This is awesome, David! It looks good! I put in a bunch of comments, and I trust you to implement the changes that you think are good and ignore any of my comments that you think aren't so great. Thanks! -Jaimie

File: client/src/app.jsx (hunk 1 above)
Comment (EJLEE808) [reply client/src/app.jsx]: Checking

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//     },
+//     ecmaVersion: 12,
+//     sourceType: 'module',
+//   },
+//   plugins: [
+//     'react',
+//   ],
+//   rules: {
+//   },
+// }
```
[the remark this answers is not in the document]
Comment (EJLEE808) [reply .eslintrc.js:23]: Deleted

File: .eslintrc.json [hunk 4]
```diff
@@ -16,8 +16,9 @@
         "ecmaFeatures": {
             "jsx": true
         },
-        "ecmaVersion": 2018,
+        "ecmaVersion": 2020,
         "sourceType": "module"
+        // "allowImportExportEverywhere": true
```
[the remark this answers is not in the document]
Comment (EJLEE808) [reply .eslintrc.json:21]: Checked

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,4 @@
 node_modules
 package-lock.json
-config.js
\ No newline at end of file
+config.js
+bundle.js
```
[the remark this answers is not in the document]
Comment (EJLEE808) [reply .gitignore:4]: Checked. I'd like to talk about this more later, but deleted for now.

File: client/src/components/overview/components/features.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    this.state = {
+      value: '',
+      features: '',
+    };
+  }
+
+  makeFeatures() {
+    let featuresArr = [];
+    for (let feature of this.props.features) {
+      console.log('feature', feature)
```
[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/src/components/overview/components/features.jsx:16]: Checked

File: client/src/components/overview/components/features.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      featuresArr.push(<div id="feature">{feature.feature} {feature.value}</div>)
+    }
+    return <div id="features">{featuresArr}</div>
+  }
+
+  componentDidMount() {
+    console.log('OUR ARRAY', this.props.features)
```
[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/src/components/overview/components/features.jsx:23]: Checked

[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/src/components/overview/components/features.jsx:23]: I commented it out

File: client/src/components/overview/overview.jsx [hunk 8]
```diff
@@ -15,6 +21,11 @@ class Overview extends React.Component {
     axios.get('http://localhost:4000/overview/product')
       .then(productInfo => {
         console.log('info', productInfo)
```
[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/src/components/overview/overview.jsx:23]: Checked

File: client/src/components/overview/overview.jsx [hunk 9]
```diff
@@ -15,6 +21,11 @@ class Overview extends React.Component {
...
+        this.setState({
+          info: productInfo.data[0],
+          styles: productInfo.data[1]
+        })
+        console.log('state', this.state.styles)
```
[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/src/components/overview/overview.jsx:28]: Checked

File: client/src/components/overview/overview.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/src/components/overview/overview.jsx:45]: I couldn't open the link :(
I think your way is much better to be honest. I just got one way to work. Lifestyle hooks ftw!

[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/src/components/overview/overview.jsx:45]: I couldn't open the link :(
I think your way is much better to be honest. I just got one way to work. Lifestyle hooks ftw!

Comment (EJLEE808) [reply client/src/components/overview/overview.jsx:45]: I couldn't open the link :(
I think your way is much better to be honest. I just got one way to work. Lifestyle hooks ftw!

**Pull request merged**